framework/src/hdgbcs/DiffusionHDGDirichletBC.C

```diff
@@ -0,0 +1,49 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "DiffusionHDGDirichletBC.h"
+#include "MooseVariableFE.h"
+#include "MooseVariableScalar.h"
+#include "Function.h"
+#include "DiffusionHDGKernel.h"
+
+registerMooseObject("MooseApp", DiffusionHDGDirichletBC);
+
+InputParameters
+DiffusionHDGDirichletBC::validParams()
+{
+  auto params = HDGIntegratedBC::validParams();
+  params += DiffusionHDGInterface::validParams();
+  params.addClassDescription("Weakly imposes Dirichlet boundary conditions for a "
+                             "hybridized discretization of diffusion equation");
+  params.addParam<FunctionName>("function", 0, "The Dirichlet value for the diffusing specie");
+  return params;
+}
+
+DiffusionHDGDirichletBC::DiffusionHDGDirichletBC(const InputParameters & parameters)
+  : HDGIntegratedBC(parameters),
+    DiffusionHDGInterface(this, this, _sys, _aux_sys, _tid),
+    _dirichlet_val(getFunction("function"))
+{
+}
+
+void
+DiffusionHDGDirichletBC::onBoundary()
+{
+  resizeData(*this);
+
+  // qu, u
```

Comment:
Linking to the paper with a comment might be important here to know what qu and u are. Same for others below.